pages/Graph/Overview.md

```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
+1. Explain how to retrieve and resolve the graph
+1. Facilitate use of SDK and interpretation of graph-data
 
 ## Assumptions
+- All graph messages will be of type `GraphChange`.
+- All `GraphChange` messages will be added to batch files for an announcement on block chain
 
-TBD
+##Terminology
+
+#### Friendship
+
+There is no concept of "friendship" within the DSNP social network graph. Friendship requires a mutual acknowledgement between 2 different DSNP Identities. 
+Friendship can be thought as "mutual following" - where 2 DSNP identities are following each other. 
+
+#### Follow
+
+A "follow" is the act of publicly following a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in adding this DSNP Id to a user's social graph.
+
+#### Unfollow
+
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in the removal of this DSNP Id from a user's social graph
+
+### Graph
 
-## Details
+The collection of DSNP identities that a given user is following via the DSNP protocol. 
 
+#### Public
+Publicly following or unfollowing a DSNP Id implies that anyone looking for these follow or unfollow events on the blockchain will
+be able to determine who the follower(DSNP id doing the following) and followee(DSNP Id being followed) are for a given event.
+
+## Graph Change Announcement Format
+Anytime a user "follows" or "unfollows" someone - a graph change event is created.
+A graph change will contain the following information: `fromId`, `changeType`, `objectId`, `nonce`. See [Messages Overview](/Messages/Overview) for more information on these fields.
+
+## Graph Storage
+All graph change events are signed and added to a [batch file](/Batches/Overview).
+Once that batch file is complete, it is [announced](/Messages/Announce) on the blockchain per the DSNP specification. 
+Each announce event specifies the type of ([dsnpType](/Messages/Announce)) messages that are in the batch file. 
+
+## Graph Retrieval, Ordering & Reading
+To have an accurate representation of the social graph for a given DSNP Identity - it is necessary to retrieve the entire graph from the chain.
+
+To retrieve the graph - do the following:
+1. Retrieve all the log events[dsnpType](/Messages/Types)`GraphChange` from the chain
+1. Retrieve the batch file from each log event. Each log event of type [GraphChange](/Messages/Announce) has a field called `dsnpURI` which contains a uri pointing to a [batch file](/Batches/Overview). 
+1. Query the batch files for the data for a particular DSNP Id to retrieve information about the respective graph. For more on how batch file storage and how to query the batch file see - [batches overview](/Batches/Overview)
+1. Order the retrieved data based on the following  
+    1. Block Number Ascending
+    1. Transaction Index Ascending
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
```

Comment:
Should all these bullet numbers be different too?

Reply:
markdown will generate consecutive numbers if you follow this format.